rewriter/DefDelegator.cc

```diff
@@ -47,26 +63,61 @@ vector<ast::ExpressionPtr> DefDelegator::run(core::MutableContext ctx, const ast
         methodName = alias->asSymbol(ctx);
     }
 
-    // sig {params(arg0: T.untyped, blk: Proc).returns(T.untyped)}
-    auto sigArgs = ast::MK::SendArgs(ast::MK::Symbol(loc, core::Names::arg0()), ast::MK::Untyped(loc),
-                                     ast::MK::Symbol(loc, core::Names::blkArg()),
-                                     ast::MK::Nilable(loc, ast::MK::Constant(loc, core::Symbols::Proc())));
+    generateStub(methodStubs, loc, methodName);
 
-    methodStubs.push_back(ast::MK::Sig(loc, std::move(sigArgs), ast::MK::Untyped(loc)));
+    // Include the original call to def_delegator so sorbet will still type-check it
+    // and throw errors if the class (or its parent) didn't `extend Forwardable`
+    methodStubs.push_back(send->deepCopy());
 
-    // def $methodName(*arg0, &blk); end
-    ast::MethodDef::ARGS_store args;
-    args.emplace_back(ast::MK::RestArg(loc, ast::MK::Local(loc, core::Names::arg0())));
-    args.emplace_back(ast::make_expression<ast::BlockArg>(loc, ast::MK::Local(loc, core::Names::blkArg())));
+    return methodStubs;
+}
 
-    methodStubs.push_back(
-        ast::MK::SyntheticMethod(loc, loc, methodName, std::move(args), ast::MK::RaiseUnimplemented(loc)));
+/// Handle #def_delegators for zero or more delegate methods
+vector<ast::ExpressionPtr> runDefDelegators(core::MutableContext ctx, const ast::Send *send) {
+    vector<ast::ExpressionPtr> methodStubs;
+    auto loc = send->loc;
 
-    // Include the original call to def_delegator so sorbet will still type-check it
+    // This method takes 1.. positional arguments and no keyword args:
+    // `def_delegators(accessor, methods*)`
+    if (!(send->numPosArgs >= 1)) {
```

Comment:
Nit: I think this is more idiomatic as `send->numPosArgs == 0`.

Reply:
Oh yes, just amended my commit to make that change and force-pushed it back up.